pytorch_lightning/strategies/launchers/subprocess_script.py

```diff
@@ -156,3 +153,87 @@ def _check_can_spawn_children(self) -> None:
                 " Possible reasons: 1) LOCAL_RANK environment variable was incorrectly modified by the user,"
                 " 2) `ClusterEnvironment.creates_processes_externally` incorrectly implemented."
             )
+
+
+class _HydraSubprocessScriptLauncher(_SubprocessScriptLauncher):
+    """Hydra Launcher to support Hydra commands."""
+
+    def _get_complete_path(self, command: str) -> str:
+        return to_absolute_path(command)
+
+    def _get_launch_command(self, command: List[str], local_rank: int) -> Tuple[List[str], Optional[str]]:
+        """Modifies the command to support Hydra initiated processes."""
+        if not HydraConfig.initialized():
+            return command, None
+
+        # If Hydra is initialized:
+        #   1) Set `cwd` to the hydra working directory
+        #   2) Use the stored configuration in `hydra_cfg.output_subdir / config.yaml` to spawn a new child
+
+        cwd = os.getcwd()
+        os_cwd = f'"{cwd}"'  # this is needed to handle characters like `=` in the directory name
+
+        hydra_cfg = HydraConfig.get()
+        hydra_output = os.path.join(cwd, hydra_cfg.output_subdir)
+
+        if __main__.__spec__ is None:  # pragma: no-cover
+            command_no_args = command[:2]
+        else:
+            # this fails for `python -m pdb -m a.b.c <args>`
+            command_no_args = command[:3]
+
+        command = command_no_args
+
+        # run the Hydra job using the current job configuration
+        # - typically located in:
+        #        RUN MODE: hydra.run.dir/.hydra/config.ayml
+        #        MULTIRUN MODE: hydra.sweep.dir/hydra.sweep.subdir/.hydra/config.yaml
+        command += ["-cp", hydra_output, "-cn", "config.yaml"]
```

Comment:
Using `-cn config.yaml` will produce an untyped config, i.e. a config (DictConfig or ListConfig) that is not backed by a dataclass.
This may or may not be a problem, depending on the use case. For example, using an untyped config means that `Enum` values will be treated by OmegaConf as strings (rather than as values of the given Enum subclass), and structured-config methods such as `OmegaConf.to_object` will not produce the expected result.

This is related to [hydra#1805](https://github.com/facebookresearch/hydra/issues/1805).

Reply:
I added support to use `experimental-rerun`